File: project5/src/main.py
import cv2

IMAGE_PATH = "./img"

def main():
    img = cv2.imread(f"{IMAGE_PATH}/1c8192x8192.bmp", cv2.IMREAD_COLOR)
    if img is None:
        print("Image not found or could not be read.")
        return
    cv2.imwrite(f"{IMAGE_PATH}/1c8192x8192copy.bmp", img)

    img = cv2.imread(f"{IMAGE_PATH}/1c4096x4096.bmp", cv2.IMREAD_COLOR)
    if img is None:
        print("Image not found or could not be read.")
        return
    cv2.imwrite(f"{IMAGE_PATH}/1c4096x4096copy.bmp", img)
    
    img = cv2.imread(f"{IMAGE_PATH}/3c4096x4096.bmp", cv2.IMREAD_COLOR)
    if img is None:
        print("Image not found or could not be read.")
        return
    cv2.imwrite(f"{IMAGE_PATH}/3c4096x4096copy.bmp", img)

    img = cv2.imread(f"{IMAGE_PATH}/4c4096x4096.bmp", cv2.IMREAD_COLOR)
    if img is None:
        print("Image not found or could not be read.")
        return
    cv2.imwrite(f"{IMAGE_PATH}/4c4096x4096copy.bmp", img)

if __name__ == "__main__":
    main()

File: project5/src/main.rs
use image::{ImageReader};

const IMAGE_DIR: &str = "img/";

fn main() {
    let img = ImageReader::open(format!("{}1c8192x8192.bmp", IMAGE_DIR))
        .expect("Failed to open image")
        .decode()
        .expect("Failed to decode image")
        .to_rgb8();

    // write the image to a new file
    img.save(format!("{}1c8192x8192_copy.bmp", IMAGE_DIR))
        .expect("Failed to save image");

    let img = ImageReader::open(format!("{}1c4096x4096.bmp", IMAGE_DIR))
        .expect("Failed to open image")
        .decode()
        .expect("Failed to decode image")
        .to_luma8();
    // write the image to a new file
    img.save(format!("{}1c4096x4096_copy.bmp", IMAGE_DIR))
        .expect("Failed to save image");

    let img = ImageReader::open(format!("{}3c4096x4096.bmp", IMAGE_DIR))
        .expect("Failed to open image")
        .decode()
        .expect("Failed to decode image")
        .to_rgb8();
    // write the image to a new file
    img.save(format!("{}3c4096x4096_copy.bmp", IMAGE_DIR))
        .expect("Failed to save image");
    let img = ImageReader::open(format!("{}4c4096x4096.bmp", IMAGE_DIR))
        .expect("Failed to open image")
        .decode()
        .expect("Failed to decode image")
        .to_rgba8();
    // write the image to a new file
    img.save(format!("{}4c4096x4096_copy.bmp", IMAGE_DIR))
        .expect("Failed to save image");
}
